### ISSUE TITLE
I cannot see any jsx a11y errors when using with react and typescript

### ISSUE THREAD

**Issue by inspiraller** [jsx-eslint/eslint-plugin-jsx-a11y#854]
I have spent the best part of the day trying to figure out why it won't display eslint error messages against this code:

Example component:
```typescript
import React, { FC } from 'react';
import { Comp } from './Comp';

const App: FC = () => (
  <div id="app">
    hello <Comp />
    <input type="text" value="something" />
  </div>
);

export default App;
```

.eslintrc.js config
```js
module.exports = {
  root: true,
  env: {
    es6: true,
    browser: true,
    jest: true,
    mocha: true,
    node: true,
  },
  parser: '@typescript-eslint/parser',
  parserOptions: {
    ecmaVersion: 2020,
    sourceType: 'module',
    ecmaFeatures: {
      jsx: true,
      modules: true,
      impliedStrict: true,
      experimentalObjectRestSpread: true,
    },
  },
  plugins: ['react', '@typescript-eslint', 'prettier', "jsx-a11y"],
  extends: [
    'airbnb',
    'plugin:react/recommended',
    'plugin:prettier/recommended',
    'plugin:@typescript-eslint/recommended',
    'plugin:import/typescript',
    'prettier',
    "plugin:jsx-a11y/recommended"
  ],
...
}
```
I have created a public git repo just for this usecase.

If you feel inclined to clone and try it for yourself:

https://github.com/inspiraller/eslint-prettier-typescript/blob/master/src/App.tsx

When loading in vscode there are NO eslint warnings or errors. When running this on the command line there are NO eslint errors:

npx eslint src/*
I would expect to see errors around the use of without adding any aria, or associated label or other accessible functionality.

**Comment by ljharb** [comment]
Your repro repo has an invalid dependency graph, and fails to install on npm 7+.

When I update eslint-config-airbnb, and all of its peer deps, one issue may be that you're missing `--ext`. However, when I run `npx eslint --ext=ts,tsx,js,jsx src/*.tsx`, `npx eslint --ext=ts,tsx,js,jsx src/*`, or `npx eslint --ext=ts,tsx,js,jsx .`, I see no errors.

The issue is that the label-has-associated-control rule only fails on *labels*. If you add config for `control-has-associated-label`, then I do see an error on the `input` in `src/App.tsx` (even with `npx eslint .`, the ideal command)

**Comment by inspiraller** [comment]
@ljharb ah yes that makes sense. Thankyou.  Yes I confirm the label error does work if I add a label and don't associate it with an input. Though it seems more useful to warn the user that they have an input without a label. When a developer adds a label in the first place its because they have some idea about accessibility and doesn't need to be reminded as much as developer who just adds an input.

**Comment by inspiraller** [comment]
I've just gone over the rule and cannot see anywhere detecting whether the htmlFor="id" actually matches an input id.
The rule only checks that the attribute exists. It is not testing it matches the actual input id.

This should fail:
```
    <label htmlFor="whatever">some label</label>
```
    
 This should be valid
```
     <label htmlFor="whatever">some label</label>
      <input id="whatever" />
```

**Comment by inspiraller** [comment]
Most websites I have worked on do not want to use labels at all. Its unnecessary boilerplate   At the very least an aria-labelled-by would be a better solution.

**Comment by ljharb** [comment]
It’s quite necessary for accessibility, on all websites. ARIA attributes are only ever a last resort for when semantic HTML is impossible.
